### PR TITLE
Move che-theia-hosted-plugin-manager-extension to che-theia

### PR DESCRIPTION
**Pull request by evidolob**
For eclipse/che#12306
Signed-off-by: Yevhen Vydolob <yvydolob@redhat.com>

### CODE REVIEW

File: extensions/che-theia-hosted-plugin-manager-extension/LICENSE [hunk 1]
```diff
@@ -0,0 +1,277 @@
+Eclipse Public License - v 2.0
```
Comment (benoitf) [inline extensions/che-theia-hosted-plugin-manager-extension/LICENSE]: I'm not sure we need this file as we have already in root folder LICENSE

File: extensions/che-theia-hosted-plugin-manager-extension/src/node/che-workspace-hosted-plugin-uri-post-processor.ts [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+
+    constructor() {
+        const restAPIConfig: IRestAPIConfig = {};
+        restAPIConfig.baseUrl = process.env.CHE_API;
+        const token = process.env.CHE_MACHINE_TOKEN;
+        if (token) {
+            restAPIConfig.headers = {};
+            restAPIConfig.headers['Authorization'] = token;
+        }
+        this.restApiClient = WorkspaceClient.getRestApi(restAPIConfig);
```
Comment (benoitf) [inline extensions/che-theia-hosted-plugin-manager-extension/src/node/che-workspace-hosted-plugin-uri-post-processor.ts]: `@eclipse-che/plugin` is not providing now the access to workspace ?

Comment (evidolob) [reply extensions/che-theia-hosted-plugin-manager-extension/src/node/che-workspace-hosted-plugin-uri-post-processor.ts]: `@eclipse-che/plugin` provide API for plugins, `che-theia-hosted-plugin-manager-extension` is extension, so no it not. But I can reuse `@eclipse-che/theia-plugin-ext` to get current workspace

**Comment by evidolob** [comment]
ci-build

**Comment by evidolob** [comment]
ci-build

**Pull request merged**